sources/packages/backend/apps/api/src/route-controllers/student-account-applications/_test_/e2e/student-account-application.aest.controller.approveStudentAccountApplication.e2e-spec.ts

```diff
@@ -0,0 +1,272 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeStudentAccountApplication,
+  createFakeUser,
+  getProviderInstanceForModule,
+  saveFakeSFASIndividual,
+} from "@sims/test-utils";
+import { AppAESTModule } from "../../../../app.aest.module";
+import { FormNames, FormService } from "../../../../services";
+import {
+  Notification,
+  NotificationMessageType,
+  StudentAccountApplication,
+  User,
+} from "@sims/sims-db";
+import { In, IsNull } from "typeorm";
+import * as faker from "faker";
+
+describe("StudentAccountApplicationAESTController(e2e)-approveStudentAccountApplication", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let sharedFormService: FormService;
+  const TEST_SIN1 = "534012702";
+  const TEST_SIN2 = "534012703";
+  const BLANK_SIN = "000000000";
+  const TEST_BIRTH_DATE1 = "2001-01-31";
+  const TEST_BIRTH_DATE2 = "2001-01-05";
+  const TEST_EMAIL = "dummy@some.domain";
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    sharedFormService = await getProviderInstanceForModule(
+      module,
+      AppAESTModule,
+      FormService,
+    );
+
+    // Insert a fake email contact to send ministry email.
+    await db.notificationMessage.update(
+      { id: NotificationMessageType.PartialStudentMatchNotification },
+      { emailContacts: [TEST_EMAIL] },
+    );
+  });
+
+  beforeEach(async () => {
+    jest.resetAllMocks();
+    // Ensure the SIN used for this method will not conflict.
+    await db.sfasIndividual.update(
+      { sin: In([TEST_SIN1, TEST_SIN2]) },
+      { sin: BLANK_SIN },
+    );
+    await db.sinValidation.update(
+      { sin: In([TEST_SIN1, TEST_SIN2]) },
+      { sin: BLANK_SIN },
+    );
+    await db.notification.update(
+      {
+        dateSent: IsNull(),
+        notificationMessage: {
+          id: NotificationMessageType.PartialStudentMatchNotification,
+        },
+      },
+      {
+        dateSent: new Date(),
+      },
+    );
+  });
+
+  function createFakeSubmittedData(user: User) {
+    return {
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+      dateOfBirth: TEST_BIRTH_DATE1,
+      phone: "1234567890",
+      sinNumber: TEST_SIN1,
+      mode: "aest-account-approval",
+      identityProvider: "bceidboth",
+      sinConsent: true,
+      gender: "X",
+      addressLine1: "address 1",
+      city: "Victoria",
+      country: "Canada",
+      postalCode: "H1H1H1H",
+      provinceState: "BC",
+      selectedCountry: "canada",
+      canadaPostalCode: "H1H1H1H",
+    };
+  }
+
+  async function mockFormioResponse(
```

Comment:
This method does not perform what is defined in its name.
Can you please move the endpoint, token, and API back to the E2E case scanrios? It would be aligned with the other E2E test cases.

Reply:
updated